src/hsl_ma57.jl

```diff
@@ -307,9 +330,24 @@ for (fname, typ) in ((:ma57a_, Float32), (:ma57ad_, Float64))
 end
 
 ## factorize -------------------------------------------------------------------
+"""
+# Factorize `Ma57` object in-place.
+
+    ma57_factorize!(M)
+
+## Input arguments:
+
+* `M::Ma57`: `Ma57` object
+
+## Return values:
+
+* `M::Ma57`: `Ma57` object
+"""
```

Comment:
```suggestion

## Example:
```JULIA
julia> using HSL
julia> T = Float64;
julia> rows = Int32[1, 2, 3, 5, 3, 4, 5]; cols = Int32[1, 1, 2, 2, 3, 3, 5];
julia> vals = T[2, 3, 4, 6, 1, 5, 1];
julia> A = sparse(rows, cols, vals);
julia> M = Ma57(A)
julia> ma57_factorize!(M)      ## factorize `Ma57` object in place
"""
```